setup.sh

```diff
@@ -7,9 +7,12 @@ echo "Setting environment variable SPACK_STACK_DIR to ${SPACK_STACK_DIR}"
 
 source ${SPACK_STACK_DIR:?}/spack/share/spack/setup-env.sh
 echo "Sourcing spack environment ${SPACK_STACK_DIR}/spack/share/spack/setup-env.sh"
+# Avoid using ~/.spack directory for caching, bootstrap, etc.
 if [ "$(uname)" != "Darwin" ]; then
   export SPACK_USER_CACHE_PATH=$SPACK_ROOT/user_cache
 fi
```

Comment:
I tried the test from the comment above (https://github.com/JCSDA/spack-stack/pull/1352#issuecomment-2427500838) and it completed on my Mac, but the `spack bootstrap now` command rebuilt my $HOME/.spack directory. I'm guessing that is because of this code here. Wouldn't it be okay to move the `.spack` directory on the Mac, or does that create other issues?